padd.sh

```diff
@@ -267,11 +269,43 @@ GetSystemInformation() {
   fi
 
   # Device model
-  if [ -f /sys/firmware/devicetree/base/model ]; then
+  if [ -f /sys/devices/virtual/dmi/id/product_name ] || [ -f /sys/devices/virtual/dmi/id/product_family ]; then
     # Get model, remove possible null byte
+    product_name=$(tr -d '\0' < /sys/devices/virtual/dmi/id/product_name)
+    product_family=$(tr -d '\0' < /sys/devices/virtual/dmi/id/product_family)
+    sys_model="$(echo "$product_name" | grep "$product_family")"
+
+    # If product_family is not contained in product_name, both are shown
+    if [ -z "$sys_model" ]; then
+      sys_model="${product_family} ${product_name}"
+    fi    
+
+  elif [ -f /sys/firmware/devicetree/base/model ]; then
     sys_model=$(tr -d '\0' < /sys/firmware/devicetree/base/model)
-  else
-    sys_model=""
+  elif [ -f /sys/devices/virtual/dmi/id/board_vendor ] || [ -f /sys/devices/virtual/dmi/id/board_name ]; then
+    sys_model=$(tr -d '\0' < /sys/devices/virtual/dmi/id/board_vendor)
+    sys_model="$sys_model $(tr -d '\0' < /sys/devices/virtual/dmi/id/board_name)"
+  elif [ -f /tmp/sysinfo/model ]; then
+    sys_model=$(tr -d '\0' < /tmp/sysinfo/model)
+  fi
+
+  # Cleaning device model from useless OEM information
+  sys_model=$(echo "$sys_model" | sed -e 's:To be filled by O.E.M.::g')
```

Comment:
Stickler test (actually Shellcheck) is complaining here. It is asking to use this variable substituion `${variable//search/replace}`, but I'm almost sure this is not POSIX.

You can try to use shell "pattern filtering":

```suggestion
  sys_model="${sys_model#"To be filled by O.E.M."}"
```

Reply:
Thanks! This is a more elegant way to clean the string.

Reply:
This only works if the searched string is on the beginning of `$sys_model`. If you need to remove from the end, use `${sys_model%"To be filled by O.E.M."}`

Reply:
Okay, so for each string there must be two filters, one for the beginning and one for the end since there might be the possibility that the value of `sys_model` is `To be filled by O.E.M. To be filled by O.E.M.`, as shown by @DL6ER